src/engine/input/key_mapping/InputEventItem.cs

```diff
@@ -260,6 +263,12 @@ private void OnKeybindingSuccessfullyChanged()
 
         // Update the button text
         UpdateButtonText();
+
+        // Consume current input even so it is only used for rebinding
+        GetTree().SetInputAsHandled();
```

Comment:
This is unnecessary because CheckNewKeyConflicts always gets called before OnKeybindingSuccessfullyChanged. So this logic is duplicate and can be removed.

Reply:
The code path where the key binding is successfully changed does not consume the event.
Currently we only consume the event in CheckNewKeyConflicts when the conflict dialog is shown.
When no conflict is found and we "return false" there I don't think we should consume the event.
Only thing I'm not sure about is whether we should consume on the second "return true" in CheckNewKeyConflicts.

If so that consume can also be moved.

Reply:
So something further needs to be done about this?

Reply:
Instead of this logic (current):
![grafik](https://user-images.githubusercontent.com/30809803/103285290-0297c400-49de-11eb-9aa3-0f47a2c900ad.png)
You could just do this logic:
![grafik](https://user-images.githubusercontent.com/30809803/103285351-307d0880-49de-11eb-80e7-3b7fbe1f2669.png)

Reply:
I didn't do it that way because logically the input isn't handled at that point.
Of course semantically it's the same and it's less duplicated code so I'll change it.